### ISSUE TITLE
Can't compute the logprobs of generated tokens

### ISSUE THREAD

**Issue by JianGuanTHU** [facebookresearch/llama#583]
When I set `logprobs` to True, I found that the model's output logits are like:  [-0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0]

I carefully checked the source [code](https://github.com/facebookresearch/llama/blob/main/llama/generation.py#L137-L143) and found that each target token is `-1` but not the last generated token. Is that correct, or do I miss anything? 

If it is a bug, I think just moving Line 137-143 after Line 155 can fix this bug.

**Comment by sAviOr287** [comment]
I have the same problem 
does anyone know the answer to this?

**Comment by YusukeO** [comment]
I have the same problem

**Comment by ruanslv** [comment]
Fixed in https://github.com/facebookresearch/llama/pull/647
